src/pydata_sphinx_theme/theme/pydata_sphinx_theme/layout.html

```diff
@@ -6,23 +6,8 @@
 {% endset %}
 {%- extends "basic/layout.html" %}
 {%- import "static/webpack-macros.html" as _webpack with context %}
-{# Metadata and asset linking #}
-{# Create the sidebar links HTML here to re-use in a few places #}
-{# If we have no sidebar links, pop the links component from the sidebar list #}
-{%- set sidebar_nav_html = generate_toctree_html("sidebar",
-show_nav_level=theme_show_nav_level|int,
-maxdepth=theme_navigation_depth|int,
-collapse=theme_collapse_navigation|tobool,
-includehidden=True,
-titles_only=True)
--%}
-{% if sidebar_nav_html | length == 0 %}
-  {% set sidebars = sidebars | reject("in", "sidebar-nav-bs.html") | list %}
-{% endif %}
```

Comment:
this conditional was moved later, to just before its results are actually needed.